### PR TITLE
DrivAerML: no-EMA retest of AdamW lr=5e-4 winner

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

**No-EMA should improve DrivAerML as it did TandemFoil (-24.7%) and AirfRANS (+29% regression from EMA).** The first DrivAerML baseline (#2440) used the default EMA=True. Given that EMA is catastrophically harmful in short training regimes (2 epochs), removing it should meaningfully improve the 71.35% baseline.

## Current Baseline

| Dataset | Metric | Value | PR |
|---|---|---|---|
| DrivAerML | val_primary/surface_rel_l2_pct | **71.35%** | #2440 (AdamW lr=5e-4, EMA=True, 2 epochs) |

## Design

Two runs: direct no-EMA retest + slightly lower LR:

| Run | Config | wandb_name |
|---|---|---|
| 1 | AdamW lr=5e-4, no-EMA | shoya-noema-adamw-lr5e4 |
| 2 | AdamW lr=3e-4, no-EMA | shoya-noema-adamw-lr3e4 |

## Instructions to Student

Run these **sequentially** (one at a time):

```bash
cd target/icml2026

# Run 1: AdamW lr=5e-4, no-EMA (direct comparison to #2440 baseline)
python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine_t_max 150 \
  --no-use-ema \
  --wandb_group shoya-noema-retest --wandb_name shoya-noema-adamw-lr5e4

# Run 2: AdamW lr=3e-4, no-EMA
python train.py --dataset drivaerml --optimizer adamw --lr 3e-4 --cosine_t_max 150 \
  --no-use-ema \
  --wandb_group shoya-noema-retest --wandb_name shoya-noema-adamw-lr3e4
```

**Report** `val_primary/surface_rel_l2_pct` and epochs completed. Target: beat 71.35%.

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2466 because Superseded: shoya reassigned to #2605 (DrivAerML 5L/256d).

**Comment by tcapelle** [comment]
STUDENT shoya:

## Results

### Summary

| Run | Config | val_primary/surface_rel_l2_pct | test_primary/surface_rel_l2_pct | Epochs | W&B Run |
|---|---|---|---|---|---|
| Baseline (#2440) | AdamW lr=5e-4, **EMA=True** | **71.35%** | — | 2 | kulxytfg |
| Run 1 | AdamW lr=5e-4, **no-EMA** | **64.34%** | **64.22%** | 2 | wcwy8qom |
| Run 2 | AdamW lr=3e-4, **no-EMA** | **62.35%** | **62.22%** | 2 | 9qhiw64b |

### Key Metrics

**Run 1 (AdamW lr=5e-4, no-EMA):**
- Epoch 1: val=71.60%, loss=0.968, lr=1.25e-4
- Epoch 2: val=64.34%, loss=0.806, lr=1.25e-4
- Test: 64.22%
- **Improvement over baseline: -7.01 pp (9.8% relative)**

**Run 2 (AdamW lr=3e-4, no-EMA):**
- Epoch 1: val=71.37%, loss=0.947, lr=7.5e-5
- Epoch 2: val=62.35%, loss=0.784, lr=7.5e-5
- Test: 62.22%
- **Improvement over baseline: -9.00 pp (12.6% relative)**

### Commands Used

```bash
# Run 1
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 150 \
  --no-use-ema --batch-size 2 \
  --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 500000 \
  --max-train-batches 200 \
  --agent shoya --wandb-group shoya-noema-retest --wandb-name "shoya/noema-adamw-lr5e4"

# Run 2
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset drivaerml --optimizer adamw --lr 3e-4 --cosine-t-max 150 \
  --no-use-ema --batch-size 2 \
  --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 500000 \
  --max-train-batches 200 \
  --agent shoya --wandb-group shoya-noema-retest --wandb-name "shoya/noema-adamw-lr3e4"
```

### Peak Memory / Timing
- GPU: NVIDIA RTX PRO 6000 Blackwell (98GB)
- Run 1: ~6.8 GB VRAM, 31 min wall time
- Run 2: ~6.8 GB VRAM, 30 min wall time

### What happened

**No-EMA comprehensively beats the EMA baseline on DrivAerML.** This confirms the pattern seen across all three datasets:
- TandemFoil: -24.7% from removing EMA
- AirfRANS: +29% regression FROM EMA  
- DrivAerML: -12.6% from removing EMA (this PR)

EMA is catastrophically harmful in our 2-epoch training regime. The EMA model with `ema_start_step=50` barely has time to accumulate meaningful exponential averages before training ends, and the averaged weights are strictly worse than the raw trained weights.

**Run 2 (lr=3e-4) outperforms Run 1 (lr=5e-4)** by 2 pp. The lower LR provides a more stable optimization trajectory, and without EMA's smoothing, the model benefits from a gentler learning rate.

**Note on training setup:** Used `--max-train-batches 200` (400 items/epoch at batch_size=2) to match the baseline's ~394 train steps/epoch. The DrivAerML dataset's view mechanism creates ~67K training views with 50K surface point subsampling, which is far too many for a 2-epoch regime. Used 500K eval surface points for eval (vs 50K for training) to reduce eval views while maintaining full per-case point coverage via strided partitioning.

### Suggested follow-ups

1. **Adopt no-EMA as the DrivAerML default** — EMA should be disabled (`--no-use-ema`) for all future DrivAerML experiments.
2. **LR sweep around 3e-4** — lr=3e-4 beat lr=5e-4 without EMA. Try lr=2e-4 and lr=1e-4 to find the optimum.
3. **Increase cosine_t_max** — Both runs had LR decayed to near-minimum after 200 steps (T_max=150). A larger T_max (e.g., 400) would keep the LR higher through the full training, potentially improving results.
4. **Fix DrivAerML view count for training** — The view mechanism creates 67K+ training items for 394 cases with 50K subsampling. This is excessive for random sampling (where view index is unused). Consider capping training views to 1 per case with random subsampling, matching the baseline's 394 steps/epoch naturally.
